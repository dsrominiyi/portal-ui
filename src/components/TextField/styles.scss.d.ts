export const textField: string;
export const inputContainer: string;
export const input: string;
export const isInvalid: string;
export const withIcon: string;
export const reverseIconPosition: string;
export const fieldIcon: string;
