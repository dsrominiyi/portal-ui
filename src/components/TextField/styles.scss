@import '../../style/vars';

.text-field {
  display: block;
}

.input-container {
  position: relative;
  display: flex;
  align-items: center;
}

.input {
  padding: 0 10px;
  height: 40px;
  width: 100%;
  border: 1px solid $brand-off-white;
  background: $brand-white;
  color: $brand-charcoal;
  font-size: 1rem;
  outline: none;
  -webkit-appearance: none;
  transition: 0.2s border;

  &.is-invalid {
    border: 1px solid $brand-danger;

    &:focus {
      border: 1px solid $brand-danger;
    }
  }

  &::placeholder {
    color: $brand-secondary;
    opacity: 1;
  }

  &::-ms-input-placeholder {
    color: $brand-secondary;
  }

  &:-ms-input-placeholder {
    color: $brand-secondary;
  }

  &:focus {
    border: 1px solid $brand-primary;
  }

  &.with-icon {
    padding: 0 40px 0 10px;

    &.reverse-icon-position {
      padding: 0 10px 0 40px;
    }
  }
}

.field-icon {
  position: absolute;
  right: 0;
  width: 40px;
  display: inline-flex;
  justify-content: center;
  align-items: center;

  &.reverse-icon-position {
    right: unset;
  }

  svg {
    > * {
      stroke: $brand-secondary;
    }
  }
}
