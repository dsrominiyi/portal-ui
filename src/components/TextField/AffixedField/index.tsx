import React, { FunctionComponent, ChangeEvent } from 'react';

import TextField from '..';

import styles from './styles.scss';

interface Props {
  prefix?: string;
  suffix?: string;
  value?: string;
  onChange?(event: ChangeEvent<HTMLInputElement>): void;
}

export const AffixedField: FunctionComponent<Props> = ({
  prefix,
  suffix,
  value,
  onChange
}: Props) => {
  return (
    <div className={styles.affixedFieldWrapper}>
      <div className={styles.prefix}>{prefix}</div>
      <TextField onChange={onChange} value={value} />
      <div className={styles.suffix}>{suffix}</div>
    </div>
  );
};

AffixedField.displayName = 'AffixedField';
export default AffixedField;
