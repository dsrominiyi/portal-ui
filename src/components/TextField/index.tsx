import React, { FunctionComponent, ChangeEvent, ReactNode } from 'react';
import classnames from 'classnames/bind';

import formStyles from '../../forms/styles.scss';
import styles from './styles.scss';

export interface Props {
  additionalClasses?: string[];
  label?: string;
  value?: string | number | string[];
  placeholder?: string;
  password?: boolean;
  maxLength?: number;
  icon?: ReactNode;
  reverseIconPosition?: boolean;
  isInvalid?: boolean;
  attributes?: { [attr: string]: string };
  onChange?(event: ChangeEvent<HTMLInputElement>): void;
  onBlur?(event: ChangeEvent<HTMLInputElement>): void;
  onFocus?(event: ChangeEvent<HTMLInputElement>): void;
}

const cx = classnames.bind(styles);

export const TextField: FunctionComponent<Props> = ({
  additionalClasses = [],
  password,
  label,
  value,
  placeholder,
  maxLength,
  icon,
  reverseIconPosition,
  isInvalid,
  attributes = {},
  onChange,
  onBlur,
  onFocus
}: Props) => {
  return (
    <div className={[styles.textField, ...additionalClasses].join(' ')}>
      {label && <div className={formStyles.label}>{label}</div>}
      <div className={styles.inputContainer}>
        <input
          type={password ? 'password' : 'text'}
          className={cx({ input: true, withIcon: !!icon, reverseIconPosition, isInvalid })}
          value={value}
          placeholder={placeholder}
          maxLength={maxLength}
          onChange={onChange}
          onBlur={onBlur}
          onFocus={onFocus}
          {...attributes}
        />
        {icon && <span className={cx({ fieldIcon: true, reverseIconPosition })}>{icon}</span>}
      </div>
    </div>
  );
};

TextField.displayName = 'TextField';
export default TextField;
