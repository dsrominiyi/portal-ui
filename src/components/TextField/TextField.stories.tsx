import React, { ReactNode } from 'react';
import { storiesOf } from '@storybook/react';

import UserIcon from 'uikit-icons/lib/User';
import PasswordIcon from 'uikit-icons/lib/Lock';

import TextField from '.';

const stories = storiesOf('TextField', module);

const render = (
  password?: boolean,
  label?: string,
  value?: string,
  icon?: ReactNode,
  placeholder?: string,
  isInvalid?: boolean
) => (props?: any) => (
  <TextField
    key={props && props.key}
    password={password}
    label={label}
    value={value}
    placeholder={placeholder}
    icon={icon}
    onChange={() => null}
    isInvalid={isInvalid}
  />
);

export const storyRenders = {
  emailField: render(false, 'Email Address', 'test@datamatters.co.uk', <UserIcon />),
  passwordField: render(true, 'Password', 'testpass', <PasswordIcon />),
  withPlaceholder: render(false, undefined, undefined, undefined, 'Placeholder text'),
  invalid: render(false, undefined, undefined, undefined, 'Placeholder text', true)
};

stories.add('Email field', storyRenders.emailField);
stories.add('Password', storyRenders.passwordField);
stories.add('With placeholder', storyRenders.withPlaceholder);
stories.add('Invalid', storyRenders.invalid);
