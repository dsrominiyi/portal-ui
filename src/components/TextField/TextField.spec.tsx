import { storyRenders } from './TextField.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('TextField', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
