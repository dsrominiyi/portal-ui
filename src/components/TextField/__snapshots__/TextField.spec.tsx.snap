// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component story renders match snapshots: emailField 1`] = `
<div
  className="textField"
>
  <div
    className="label"
  >
    Email Address
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="input withIcon"
      onChange={[Function]}
      type="text"
      value="test@datamatters.co.uk"
    />
    <span
      className="fieldIcon"
    >
      <SvgUser />
    </span>
  </div>
</div>
`;

exports[`TextField component story renders match snapshots: invalid 1`] = `
<div
  className="textField"
>
  <div
    className="inputContainer"
  >
    <input
      className="input isInvalid"
      onChange={[Function]}
      placeholder="Placeholder text"
      type="text"
    />
  </div>
</div>
`;

exports[`TextField component story renders match snapshots: passwordField 1`] = `
<div
  className="textField"
>
  <div
    className="label"
  >
    Password
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="input withIcon"
      onChange={[Function]}
      type="password"
      value="testpass"
    />
    <span
      className="fieldIcon"
    >
      <SvgLock />
    </span>
  </div>
</div>
`;

exports[`TextField component story renders match snapshots: withPlaceholder 1`] = `
<div
  className="textField"
>
  <div
    className="inputContainer"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Placeholder text"
      type="text"
    />
  </div>
</div>
`;
